### PR TITLE
A couple fixes in the WCIF JSON Schema

### LINKED ISSUE #3059: Remove scrambleGroupCount once TNoodle does not read from it
See https://github.com/thewca/worldcubeassociation.org/pull/3058, which renames "scramble group" to "scramble set". We could not remove `scrambleGroupCount` from the WCIF, because TNoodle reads that field. However, once a version of TNoodle with <<< in it is released, this will be safe to clean up https://github.com/thewca/tnoodle/commit/9e58918eb148bda519f0fa71d560da276c808acc.

### PR DESCRIPTION
**Pull request by jonatanklosko**
Fixes #3059.

### CODE REVIEW

File: WcaOnRails/app/models/round.rb [hunk 1]
```diff
@@ -140,13 +140,6 @@ def to_wcif
       "timeLimit" => event.can_change_time_limit? ? time_limit&.to_wcif : nil,
       "cutoff" => cutoff&.to_wcif,
       "advancementCondition" => advancement_condition&.to_wcif,
-
-      # TODO: This is here for backwards compatibility with TNoodle 0.13.4,
-      # which looks at scrambleGroupCount. We can remove this once a new
-      # version of TNoodle is released which looks at a different field.
-      # See https://github.com/thewca/worldcubeassociation.org/issues/3059.
-      "scrambleGroupCount" => self.scramble_set_count,
```
Comment (jonatanklosko) [inline WcaOnRails/app/models/round.rb]: `TNoodle 1.13.5` was released 1,5 month ago, so I believe it's safe to get rid of this (and also update the TNoodle code to no longer refer to `scrambleGroupCount`).

File: WcaOnRails/app/models/round.rb [hunk 2]
```diff
@@ -161,8 +154,7 @@ def self.wcif_json_schema
         "timeLimit" => TimeLimit.wcif_json_schema,
         "cutoff" => Cutoff.wcif_json_schema,
         "advancementCondition" => AdvancementCondition.wcif_json_schema,
-        "results" => { "type" => "array", "items" => { "type" => RoundResult.wcif_json_schema } },
-        "groups" => { "type" => "array" }, # TODO: expand on this
```
Comment (jonatanklosko) [inline WcaOnRails/app/models/round.rb]: This is no longer a part of the WCIF spec.

File: WcaOnRails/app/models/round.rb [hunk 3]
```diff
@@ -161,8 +154,8 @@ def self.wcif_json_schema
...
+        "results" => { "type" => "array", "items" => RoundResult.wcif_json_schema },
+        "scrambleSets" => { "type" => "array" }, # TODO: expand on this
```
Comment (jonatanklosko) [inline WcaOnRails/app/models/round.rb]: The current WCIF spec defines `scrambleSets` instead of `groups`.

**Comment by jfly** [comment]
LGTM, thanks!

**Pull request merged**